src/NumericalAlgorithms/Spectral/SwshInterpolation.cpp

```diff
@@ -143,133 +150,174 @@ std::complex<double> SpinWeightedSphericalHarmonic::evaluate(
-      a = static_cast<double>(abs(Spin + m));
-      b = static_cast<double>(abs(Spin - m));
-      l_plus_k = static_cast<double>(-(abs(Spin + m) + b) / 2 + l_min);
-
       prefactor_accumulator =
-          sqrt((2.0 * static_cast<double>(abs(m)) + 1.0) *
-               (l_plus_k + a + b - 1.0) * (l_plus_k + a + b) /
-               ((2.0 * static_cast<double>(abs(m)) - 1.0) * (l_plus_k + a) *
-                (l_plus_k + b)));
-      lambda_difference = gsl::at(lambda, m + static_cast<int>(l_max)) + 1;
-      if (m > 0) {
-        lambda_difference -= gsl::at(lambda, m - 1 + static_cast<int>(l_max));
-      } else {
-        lambda_difference -= gsl::at(lambda, m + 1 + static_cast<int>(l_max));
-      }
-      prefactor_accumulator *= lambda_difference % 2 == 0 ? 1.0 : -1.0;
+          sqrt(prefactor_accumulator * static_cast<double>(2.0 * l_min + 1.0) /
+               (4.0 * M_PI));
+      prefactor_accumulator *=
+          ((m + gsl::at(lambda, m + static_cast<int>(l_max))) % 2) == 0 ? 1.0
+                                                                        : -1.0;
       // this is the right order of the casts, other orders give the wrong
       // answer NOLINTNEXTLINE(misc-misplaced-widening-cast)
-      harmonic_m_recurrence_prefactors[static_cast<size_t>(
+      harmonic_at_l_min_prefactors[static_cast<size_t>(
           m + static_cast<int>(l_max))] = prefactor_accumulator;
+
+      // pre-compute the prefactors for bootstrapping the second-to-lowest order
+      // harmonics for each m
+
+      // this is the right order of the casts, other orders give the wrong
+      // answer NOLINTNEXTLINE(misc-misplaced-widening-cast)
+      harmonic_at_l_min_plus_one_recurrence_prefactors[static_cast<size_t>(
+          m + static_cast<int>(l_max))] =
+          sqrt((2.0 * static_cast<double>(l_min) + 3.0) * (l_min_plus_k + 1.0) *
+               (l_min_plus_k + a + b + 1.0) /
+               ((2.0 * static_cast<double>(l_min) + 1.0) *
+                (l_min_plus_k + a + 1.0) * (l_min_plus_k + b + 1.0)));
+    }
+    // separate loop because we'll need the lambdas entirely populated for this
+    // set of prefactors
+    int lambda_difference;
+    for (int m = -static_cast<int>(l_max); m <= static_cast<int>(l_max); ++m) {
+      if (std::abs(m) > std::abs(Spin)) {
+        l_min = std::max(std::abs(m), std::abs(Spin));
+        a = static_cast<double>(std::abs(Spin + m));
+        b = static_cast<double>(std::abs(Spin - m));
+        l_min_plus_k =
+            static_cast<double>(-0.5 * (std::abs(Spin + m) + b) + l_min);
+
+        prefactor_accumulator =
+            sqrt((2.0 * static_cast<double>(std::abs(m)) + 1.0) *
+                 (l_min_plus_k + a + b - 1.0) * (l_min_plus_k + a + b) /
+                 ((2.0 * static_cast<double>(std::abs(m)) - 1.0) *
+                  (l_min_plus_k + a) * (l_min_plus_k + b)));
+        lambda_difference = gsl::at(lambda, m + static_cast<int>(l_max)) + 1;
```

Comment:
I'm confused by the `lambda(m) + 1` here compared to the documentation. I might have misunderstood the dox, though.

Reply:
This has absorbed the extra minus sign out front into the delta lambda.

Reply:
This has absorbed the extra minus sign out front into the delta lambda.

Reply:
Sorry, I think we're talking about different things. In the dox you have:

Delta lambda(m) = lambda(m) - lambda(m-1)

but the code reads as:

Delta lambda(m) = lambda(m) + 1 - lambda(m - 1)

unless there's something subtle going on with the `+ l_max`. It's this extra 1 that I'm confused about.

Reply:
Yes, sorry, that's what I mean. The recurrence relation has a factor of -1 out front, and I've omitted the -1 from the overall factor and included it in the delta lambda because the role of delta lambda is the factor of (-1)^\Delta lambda in the recurrence.
If that's too confusing, I can probably hunt down the places the -1 should go instead of here.

Reply:
Ah ok, no that makes sense. Maybe just add a comment in the C++ code?

Reply:
comment added and squashed in.